src/string_impls/custom_parsers/ss58.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2022-2023 Parity Technologies (UK) Ltd. (admin@parity.io)
+// This file is a part of the scale-value crate.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{stringify::ParseError, Value};
+
+/// Attempt to parse an ss58 address into a [`Value<()>`] (or more specifically,
+/// an unnamed composite wrapped in a newtype which represents an AccountId32).
+///
+/// - Returns `None` if we can't parse the address.
+/// - Returns `Some(value)` if parsing was successful. In this case, the string
+///   reference given is wound forwards to consume what was parsed.
+pub fn parse_ss58(s: &mut &str) -> Option<Result<Value<()>, ParseError>> {
+    let bytes = parse_ss58_bytes(s)?;
+    Some(Ok(Value::from_bytes(bytes)))
+}
+
+fn parse_ss58_bytes(s: &mut &str) -> Option<Vec<u8>> {
+    const CHECKSUM_LEN: usize = 2;
+
+    // ss58 addresses are base58 encoded. Base58 is all alphanumeric chars
+    // minus a few that look potentially similar. So, gather alphanumeric chars
+    // first.
+    let end_idx = s.find(|c: char| !c.is_ascii_alphanumeric()).unwrap_or(s.len());
+    let maybe_ss58 = &s[0..end_idx];
+    let rest = &s[end_idx..];
+
+    if maybe_ss58.is_empty() {
+        return None;
+    }
+
+    // Break early on obvious non-addresses that we want to parse elsewise, ie true, false or numbers.
+    // This is mostly an optimisation but also eliminates some potential weird edge cases.
+    if maybe_ss58 == "true"
+        || maybe_ss58 == "false"
+        || maybe_ss58.find(|c: char| c.is_ascii_alphabetic()).is_none()
```

Comment:
I would prefer `maybe_ss58.all(|c| !c.is_ascii_alphabetic())`

Reply:
Thanks; that is nicer! (I went with `maybe_ss58.chars().all(|c: char| c.is_ascii_digit())` in the end)